### PR TITLE
Remove support for legacy providers

### PR DESCRIPTION
**Pull request by hatchan**

### CODE REVIEW

File: src/tasks/service.rs [hunk 1]
```diff
@@ -509,17 +504,14 @@ impl ProxyService {
     ))]
     async fn handle_invoke_proxy_message_v1(
```
Comment (gagbo) [inline src/tasks/service.rs:505]: I think you can even delete the entire function there, and when stuff fails to compile, just continue deleting everything that mentioned this code :)

File: src/tasks/service.rs [hunk 2]
```diff
@@ -826,12 +750,8 @@ async fn load_wasm_modules(wasm_dir: &Path, provider_types: Vec<String>) -> Wasm
...
 }
-fn get_protocol_version(provider_type: &str) -> u8 {
-    if V1_PROVIDERS.contains(&provider_type) {
-        1
-    } else {
-        2
-    }
+fn get_protocol_version(_: &str) -> u8 {
+    2
 }
```
Comment (gagbo) [inline src/tasks/service.rs:755]: Is it possible to inline all the uses of that function? Now that we might have multiple versions of providers in the wild, the "protocol detection" logic should happen entirely from the communication between the provider and FPD anyways, so we should never need that `get_protocol_version(hardcoded_name: &str)` function again

**Review by arendjr — APPROVED** [review]
:+1:

File: src/tasks/service.rs (hunk 1 above)
Comment (hatchan) [reply src/tasks/service.rs:505]: I decided to keep it, since the message "invoke proxy v1" message still exists.

Comment (hatchan) [reply src/tasks/service.rs:505]: Actually it doesn't look like this is caused by a specific message. Will see how I can change it.

File: src/tasks/service.rs (hunk 2 above)
Comment (hatchan) [reply src/tasks/service.rs]: Implemented in ce7927dc145b8a4ef07cdabc47f8fc160dfe0d6b

File: src/tasks/service.rs (hunk 1 above)
Comment (hatchan) [reply src/tasks/service.rs]: Implemented in cae96d4da512658dfd05b27011f292f9d3838124

**Review by gagbo — APPROVED** [review]
LGTM!!

**Comment by hatchan** [comment]
I've also changed the changelog message to indicate dropping of the v1 format instead of the `ProviderEvent` related change.

**Pull request merged**